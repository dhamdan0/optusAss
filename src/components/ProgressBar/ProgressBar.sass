.progressBar
  border-radius: 5px
  padding-block: .5rem
  margin: 1rem
  text-align: center
  font-weight: bold
  border: 1px solid  var(--color-primary-main)
  transition: background-size .3s
  background-position: 0 0
  background-image: linear-gradient(to right,  var(--color-primary-main) 0%,  var(--color-primary-darker) 100%)
  background-repeat: no-repeat
  background-size: 0 100%

  &--error
    background-image: linear-gradient(to right,  var(--color-error) 0%,  var(--color-error) 100%)
    border-color: var(--color-error)
