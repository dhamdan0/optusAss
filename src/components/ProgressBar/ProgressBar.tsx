import './ProgressBar.sass'
import classnames from 'classnames'
import { memo } from 'react'

interface ProgressBarProps {
  progress: number
}
const cssPrefix: string = 'progressBar'

function ProgressBar ({ progress }: ProgressBarProps) {
  const normalizedProgress = Math.max(Math.round(progress), 0)
  const isExceedLimit = progress > 100

  return (
        <div
            className={
              classnames(
                  `${cssPrefix}`,
                  isExceedLimit && `${cssPrefix}--error`
              )
            }
            style={ { backgroundSize: `${normalizedProgress}% 100%` } }
        >
            <span> {`${normalizedProgress} %`}</span>
      </div>
  )
}

export default memo(ProgressBar)
