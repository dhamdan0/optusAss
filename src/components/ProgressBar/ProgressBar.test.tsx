import React from 'react'
import { render, screen } from '@testing-library/react'
import ProgressBar from './ProgressBar'

function testComponentWithProps (progress: number) {
  const container = render(
      <ProgressBar progress={progress}/>)
  expect(container).toMatchSnapshot()
}

describe('ProgressBar', () => {
  it('renders progress bar when exceed 100', () => {
    testComponentWithProps(200)
    const value = screen.getByText(/200 %/i)
    expect(value.parentElement).toHaveClass('progressBar--error')
  })

  it('renders progress bar when less that 0', () => {
    testComponentWithProps(-25)
    const zero = screen.getByText(/0 %/i)
    expect(zero).toBeInTheDocument()
  })

  it('renders progress bar when 100', () => {
    testComponentWithProps(100)
  })

  it('renders progress bar when 50', () => {
    testComponentWithProps(50)
  })
})
