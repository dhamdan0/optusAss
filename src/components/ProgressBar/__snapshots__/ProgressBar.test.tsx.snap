// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders progress bar when 50 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progressBar"
        style="background-size: 50% 100%;"
      >
        <span>
           
          50 %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progressBar"
      style="background-size: 50% 100%;"
    >
      <span>
         
        50 %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar renders progress bar when 100 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progressBar"
        style="background-size: 100% 100%;"
      >
        <span>
           
          100 %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progressBar"
      style="background-size: 100% 100%;"
    >
      <span>
         
        100 %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar renders progress bar when exceed 100 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progressBar progressBar--error"
        style="background-size: 200% 100%;"
      >
        <span>
           
          200 %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progressBar progressBar--error"
      style="background-size: 200% 100%;"
    >
      <span>
         
        200 %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar renders progress bar when less that 0 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progressBar"
        style="background-size: 0% 100%;"
      >
        <span>
           
          0 %
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progressBar"
      style="background-size: 0% 100%;"
    >
      <span>
         
        0 %
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
